### PR TITLE
Set gRPC endpoint during onboarding, part 2

### LINKED ISSUE #605: Add RPC selection to onboarding flow
Currently, Penumbra Labs runs an RPC at `grpc.testnet{,-preview}.penumbra.zone`. However, we won't be doing this for a production mainnet. We want to ensure users have choice in their RPC provider and that there isn't a single entity acting as a chokepoint.

For this reason, the extension needs to have the user select their RPC endpoint while they are onboarding onto the extension. This should happen after they input their seed phrase and configure a passphrase.  To do this, we'll need:

- [ ] JSON configuration data in the extension source with a list of endpoints. Presumably each list item should include:
  - [ ] The RPC URL itself (eg `https://grpc.testnet.penumbra.zone`)
  - [ ] A human-readable name (eg `Penumbra Labs Testnet RPC`)
  - [ ] Optionally a logo image for the RPC provider (eg the Penumbra Labs logo)
- [ ] A dialog in the user flow after they set their seed phrase that prompts them to select from the list. The order of the list should be randomized on every appearance
- [ ] The list should have a special entry at the end labeled "Custom RPC" with free-form user input for the RPC URL.
- [ ] The list should have the first item (which is random, because of order randomization) preselected, so that a user who doesn't care about the RPC selection can just immediately click "Next" without having to think about it if they don't want to.

This would also help me with flows around local devnets, switching to preview, etc; it's easier to reset the whole wallet state and be able to set the target chain without having to edit internal constants.

### PR DESCRIPTION
**Pull request by jessepinho**
This PR takes the UI from #868 and uses it in the RPC settings page of the extension. It also does a bit of refactoring to better share the functionality between the RPC settings page and the RPC onboarding page. And lastly, it shows the selected chain ID (or error message) at the bottom of the form.

https://github.com/penumbra-zone/web/assets/1121544/63f05064-a426-4e67-82f7-11dc15e65e39

https://github.com/penumbra-zone/web/assets/1121544/7192fd2b-bc64-4804-8f2a-f664eb50151d

## In this PR
- Extract a `<GrpcEndpointForm />` component that's shared between the onboarding screen and the settings screen of the extension.
- Create a `<PaddingWrapper />` component to ensure consistent padding in the various component popup screens.

Closes #605

### CODE REVIEW

File: apps/extension/src/routes/page/onboarding/routes.tsx
Comment (jessepinho) [inline apps/extension/src/routes/page/onboarding/routes.tsx:9]: Consolidating the naming

File: apps/extension/src/routes/page/onboarding/set-grpc-endpoint.tsx
Comment (jessepinho) [inline apps/extension/src/routes/page/onboarding/set-grpc-endpoint.tsx:8]: This file used to be `set-rpc-endpoint.tsx`. When I renamed it to `set-grpc-endpoint.tsx`, _and_ extracted the `GrpcEndpointForm` component, the file was so different that Git thought of it as an entirely different file.

File: apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts
Comment (jessepinho) [inline apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts:27]: This hook basically extracts all the business logic out of the `SettingsRPC` component

File: apps/extension/src/routes/popup/padding-wrapper.tsx
Comment (jessepinho) [inline apps/extension/src/routes/popup/padding-wrapper.tsx:7]: For now, this is only used in one component (the RPC settings page). Once this PR is merged, I'll make a follow-up PR to use this throughout the extension popup screens.

File: apps/extension/src/shared/components/grpc-endpoint-form/index.tsx
Comment (turbocrime) [inline apps/extension/src/shared/components/grpc-endpoint-form/index.tsx:37]: minor but i'd make this a bare form element instead of wrapping it in a div

File: apps/extension/src/routes/popup/padding-wrapper.tsx
Comment (turbocrime) [reply apps/extension/src/routes/popup/padding-wrapper.tsx:7]: it seems like a custom component is the incorrect way to do this, it should be part of a layout or other existing parent component

File: apps/extension/package.json
Comment (turbocrime) [inline apps/extension/package.json:35]: tree shaking still sucks, and lodash is about about 1.5mb. i went through some effort to remove it as a dep a while back - if we're just going to use `debounce`, i would prefer to just write a debounce fn

File: apps/extension/src/routes/page/onboarding/routes.tsx
Comment (turbocrime) [reply apps/extension/src/routes/page/onboarding/routes.tsx:9]: would prefer to align on "rpc", it's a more generic term

File: apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts
Comment (turbocrime) [inline apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts:95]: on mainnet, this is a more serious condition than is typical in our experience as developers constantly flipping between testnet/preview. there should be a confirmation or other significant ui hurdle at this point.

File: apps/extension/src/shared/components/grpc-endpoint-form/index.tsx
Comment (jessepinho) [reply apps/extension/src/shared/components/grpc-endpoint-form/index.tsx:37]: Yeah I was going to, but I actually have a single element _outside_ the form at the bottom of this component that needs to be spaced apart from the form with `gap-2.

File: apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts
Comment (jessepinho) [reply apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts:95]: On it

Comment (jessepinho) [reply apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts:102]: OK, I added a confirmation dialog in [this commit](https://github.com/penumbra-zone/web/pull/877/commits/4251dd970f81c78a9e347f51d1519f867b1e6954?diff=split&w=1).

Was the rest of this PR a ✅ ? If so, could you please approve if that latest commit looks good?

File: apps/extension/src/shared/components/grpc-endpoint-form/index.tsx
Comment (TalDerei) [inline apps/extension/src/shared/components/grpc-endpoint-form/index.tsx:87]: clicking on this link in the RPC settings page of the extension takes you to this outdated link: https://github.com/penumbra-zone/web/blob/main/packages/constants/src/grpc-endpoints.ts

**Review by turbocrime — APPROVED** [review]

File: apps/extension/src/shared/components/grpc-endpoint-form/index.tsx
Comment (jessepinho) [reply apps/extension/src/shared/components/grpc-endpoint-form/index.tsx:87]: @TalDerei that's actually the new link!

File: apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts
Comment (jessepinho) [reply apps/extension/src/shared/components/grpc-endpoint-form/use-grpc-endpoint-form.ts:28]: Yeah, you're right — I'll do a follow-up PR for that

File: apps/extension/src/routes/page/onboarding/routes.tsx
Comment (jessepinho) [reply apps/extension/src/routes/page/onboarding/routes.tsx:9]: Ugh, same. I'll merge this and then discuss in Discord. The issue was, the local storage key is `grpcEndpoint`, so I figured we should standardize on that since it's already in a bunch of users' local storage.

**Pull request merged**
